lib/mix/lib/mix/tasks/test.ex

```diff
@@ -14,17 +16,84 @@ defmodule Mix.Tasks.Test do
           Mix.raise("Failed to cover compile directory: " <> compile_path)
       end
 
-      output = opts[:output]
-
       fn ->
-        Mix.shell().info("\nGenerating cover results ...")
-        File.mkdir_p!(output)
+        Mix.shell().info("\nGenerating cover results ...\n")
+        {:result, ok, _fail} = :cover.analyse(:coverage, :module)
+
+        results = Enum.sort_by(ok, &percentage(elem(&1, 1)), &>=/2)
 
-        Enum.each(:cover.modules(), fn mod ->
-          {:ok, _} = :cover.analyse_to_file(mod, '#{output}/#{mod}.html', [:html])
+        if summary_opts = Keyword.get(opts, :summary, true), do: console(results, summary_opts)
+
+        html(results, opts)
+      end
+    end
+
+    defp console(results, true), do: console(results, [])
+
+    defp console(results, opts) when is_list(opts) do
+      Mix.shell().info("Percentage | Module")
+      Mix.shell().info("-----------|--------------------------")
+
+      total =
+        results
+        |> Stream.each(&display(&1, opts))
+        |> Enum.reduce({0, 0}, fn {_, {cov, not_cov}}, {tot_cov, tot_not_cov} ->
+          {tot_cov + cov, tot_not_cov + not_cov}
         end)
+
+      Mix.shell().info("-----------|--------------------------")
+
+      display({"Total", total}, opts)
+      Mix.shell().info("")
+    end
+
+    defp console(_, value) when value in [nil, false], do: nil
+
+    defp html(results, opts) do
+      output = opts[:output]
+      File.mkdir_p!(output)
+
+      for {mod, _} <- results do
+        {:ok, _} = :cover.analyse_to_file(mod, '#{output}/#{mod}.html', [:html])
       end
+
+      Mix.shell().info([
+        "Generated HTML coverage results in '",
+        output,
+        "' directory\n"
+      ])
+    end
+
+    defp color(percentage, true), do: color(percentage, @default_threshold)
+    defp color(_, false), do: ""
+    defp color(percentage, threshold) when percentage > threshold, do: :green
+    defp color(_, _), do: :red
+
+    defp display({name, coverage}, opts) do
+      threshold = Keyword.get(opts, :threshold, @default_threshold)
+      percentage = percentage(coverage)
+
+      Mix.shell().info([
+        color(percentage, threshold),
+        format(percentage, 9),
+        "%",
+        :reset,
+        " | ",
+        format_name(name)
+      ])
     end
+
+    defp percentage({0, 0}), do: 100
+    defp percentage({cov, not_cov}), do: cov / (cov + not_cov) * 100
+
+    defp format(num, len) when is_integer(num),
```

Comment:
Generally we use full name in arguments, so it reads better `number` and `length`

Reply:
@eksperimental my problem with `length` is that vim-elixir colours it as a keyword which I find distracting. Also I find `len` and `num` popular enough that I didn't even thought about using full name.

Reply:
> my problem with length is that vim-elixir colours it as a keyword which I find distracting

Wait what why?  It's not a keyword!  o.O

Reply:
elixir-editors/vim-elixir#378